pkg/generate/app/cmd/newapp.go

```diff
@@ -219,30 +226,92 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 	}
 }
 
+func (c *AppConfig) tryToAddEnvironmentArguments(s string) bool {
+	rc := cmdutil.IsEnvironmentArgument(s)
+	if rc {
+		glog.V(2).Infof("treating %s as possible environment argument\n", s)
+		c.Environment = append(c.Environment, s)
+	}
+	return rc
+}
+
+func (c *AppConfig) tryToAddSourceArguments(s string) bool {
+	remote, rerr := app.IsRemoteRepository(s)
+	local, derr := app.IsDirectory(s)
+
+	if remote || local {
+		glog.V(2).Infof("treating %s as possible source repo\n", s)
+		c.SourceRepositories = append(c.SourceRepositories, s)
+		return true
+	}
+
+	if rerr != nil {
+		c.ArgumentClassificationErrors = append(c.ArgumentClassificationErrors, ArgumentClassificationError{
+			Key:   fmt.Sprintf("%s as a GIT repository URL", s),
+			Value: rerr,
+		})
+	}
+
+	if derr != nil {
+		c.ArgumentClassificationErrors = append(c.ArgumentClassificationErrors, ArgumentClassificationError{
+			Key:   fmt.Sprintf("%s as a local directory pointing to a GIT repository", s),
+			Value: derr,
+		})
+	}
+
+	return false
+}
+
+func (c *AppConfig) tryToAddComponentArguments(s string) bool {
+	err := app.IsComponentReference(s)
+	if err == nil {
+		glog.V(2).Infof("treating %s as a component ref\n", s)
+		c.Components = append(c.Components, s)
+		return true
+	}
+	c.ArgumentClassificationErrors = append(c.ArgumentClassificationErrors, ArgumentClassificationError{
+		Key:   fmt.Sprintf("%s as an imagestream tag or docker image reference ", s),
```

Comment:
changed to `"%s as a template loaded in an accessible project, an imagestream tag, or a docker image reference`